core/primitives/src/block.rs

```diff
@@ -522,38 +521,59 @@ impl Block {
     }
 }
 
-#[derive(Clone)]
-pub enum MaybeNew<'a, T> {
-    New(&'a T),
-    Old(&'a T),
+#[derive(Clone, Debug)]
+pub enum ChunkType<'a> {
+    New(&'a ShardChunkHeader),
+    Old(&'a ShardChunkHeader),
 }
 
-fn annotate_chunk(
-    chunk: &ShardChunkHeader,
-    block_height: BlockHeight,
-) -> MaybeNew<ShardChunkHeader> {
-    if chunk.is_new_chunk(block_height) { MaybeNew::New(chunk) } else { MaybeNew::Old(chunk) }
+/// Implements Deref for ChunkType to allow access to ShardChunkHeader methods directly.
+impl Deref for ChunkType<'_> {
```

Comment:
Are you sure we want this? The idea behind MaybeNew was to force developers to think about different handling of old and new chunks, whereas this makes it easier to bypass this protection.

Reply:
I debated between whether it makes sense to do this or not, but while working on the changes, specially in part 4 of the PR series, I realized a lot of places don't actually need to distinguish between old and new chunks. In those cases, it becomes very inconvenient to call functions on the header as we always would need to do an unnecessary match.

For now, let's go with this simplified approach and maybe we can convert everything to iter_raw if needed?